pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -202,158 +209,142 @@ public Map<String, Integer> getCSVHeaderMap() {
 
   @Override
   public boolean hasNext() {
-    if (_useLineIterator) {
-      // When line iterator is used, the call to this method won't throw an exception. The default and the only iterator
-      // from commons-csv library can throw an exception upon calling the hasNext() method. The line iterator overcomes
-      // this limitation.
-      return readNextRecord();
-    }
     return _iterator.hasNext();
   }
 
   @Override
   public GenericRow next()
       throws IOException {
-    if (_useLineIterator) {
-      return _nextRecord;
-    } else {
-      return next(new GenericRow());
-    }
+    return next(new GenericRow());
   }
 
   @Override
   public GenericRow next(GenericRow reuse)
       throws IOException {
-    if (_useLineIterator) {
-      reuse.init(_nextRecord);
-    } else {
-      CSVRecord record = _iterator.next();
-      _recordExtractor.extract(record, reuse);
-    }
+    CSVRecord record = _iterator.next();
+    _recordExtractor.extract(record, reuse);
     return reuse;
   }
 
   @Override
   public void rewind()
       throws IOException {
-    if (_useLineIterator) {
-      resetLineIteratorResources();
-    }
-
     if (_parser != null && !_parser.isClosed()) {
       _parser.close();
     }
-
-    init();
+    closeIterator();
+    initIterator();
   }
 
   @Override
   public void close()
       throws IOException {
-    if (_useLineIterator) {
-      resetLineIteratorResources();
-    }
+    closeIterator();
 
     if (_parser != null && !_parser.isClosed()) {
       _parser.close();
     }
   }
 
-  private boolean readNextRecord() {
-    try {
-      _nextRecord = null;
-      GenericRow genericRow = new GenericRow();
-      readNextLine(genericRow);
-      _nextRecord = genericRow;
-    } catch (Exception e) {
-      LOGGER.info("Error parsing next record.", e);
+  private void closeIterator()
+      throws IOException {
+    // if header is not provided by the client it would be rebuilt. When it's provided by the client it's initialized
```

Comment:
We dont need this logic that I see in resetLineIteratorResources ?

 _nextLine = null;

    LOGGER.info("Total lines skipped in file: {} were: {}", _dataFile, _skippedLinesCount);
    _skippedLinesCount = 0;

Reply:
`_skippedLinesCount` is not used anywhere other than logging. I think this ideally should be part of something like `recordReaderStats` that needs to come in through the interface. Then we can summarize, aggregate and escalate as metrics. Currently with a line of log, I just felt it wasn't adding a lot of value.

Restoring this back meant casting `Iterator<>` to `LineIterator<>` or persisting it in the parent class at top level. I'd rather introduce a container for this and have readers publish metrics to it.

thoughts?